components/SanityBlockContentNext.tsx

```diff
@@ -102,15 +103,30 @@ const serializers = {
                 </Link>
             );
         },
+        interpolate: function renderInterpolate({mark}) {
+            const context = useContext(SanityContext);
+            const {prop} = mark;
+
+            const value = context[prop];
+
+            return <>{value}</>;
+        },
     },
 };
 
-export const SanityBlockContent = (props) => {
+const SanityContext = React.createContext({});
+
+export const SanityBlockContent = (props: {
+    blocks: any;
+    templateProps?: object;
+}) => {
     return (
-        <BlockContent
-            blocks={props.blocks}
-            serializers={serializers}
-            {...client.config()}
-        />
+        <SanityContext.Provider value={props.templateProps}>
```

Comment:
SanityBlockContent lar oss ikke sende med customProps til renderfunksjonen. Bruker derfor React.Context for å holde på verdier fra `getStaticProps` etc som skal interpoleres med tekster fra Sanity.